examples/echo_server.hs

```diff
@@ -0,0 +1,18 @@
+{-# OPTIONS -XOverloadedStrings #-}
+
+import qualified Network.Freddy as Freddy
+import Data.ByteString.Lazy.Char8 (ByteString)
+
+processMessage :: Freddy.Request -> IO ()
+processMessage (Freddy.Request body replyWith failWith) = replyWith body
+
+main = do
+  fConnection <- Freddy.connect "127.0.0.1" "/" "guest" "guest"
```

Comment:
**See Urmas response instead**,

My little IO explanation (probably more useful to google instead though):

Freddy.connect does IO. This means that it must return `IO Connection`. There's no way to just return `Connection`. This is how haskell keeps pure functions separate.

Imagine a case
```haskell
-- a function that fetches an operator from a database
fetchOperator :: Int -> IO Operator

-- this one doesn't deals with pure Operator, not IO Operator
showOperatorName :: Operator -> String

main :: IO ()
main = do
  -- we are in IO monad

  -- so we can do an io operator and extract the result
  operator <-fetchOperator 5

  -- now we can call a pure function and the output it in IO again
  putStrLn $ showOperatorName operator
```

hopefully I didn't confuse people even more :)